src/main/kotlin/com/cognifide/gradle/aem/pkg/PackagePlugin.kt

```diff
@@ -50,6 +51,9 @@ class PackagePlugin : Plugin<Project> {
         activate.mustRunAfter(compose, upload, install)
 
         deploy.dependsOn(compose)
+
+        download.dependsOn(clean)
```

Comment:
that's wrong. terribly wrong. Gradle tasks should be incremental. running clean by download will incidentally force to recompile java etc. there could be `mustRunAfter` relation only.

clean task should be always called exclusively.  your download task should clean previous results if they are created / ensure that it is rerunable without clean

as an example think of `aemCreate` . it is removing dir in which aem will be extracted as first step it is treating previous results/files as garbage until lock file being created in the end of that process.

Reply:
Ok, I'll try to omit the clean task. 
It was added there mainly because the "shell" package is build form the outputs of aemPrepare. Clean was used to avoid any 'stale' content be used in package creation. Most likely it's not strictly necessary.

Reply:
Removed the clean dependency. It wasn't necessary